sqlalchemy_ibmi/base.py

```diff
@@ -749,49 +749,19 @@ def create_connect_args(self, url):
         opts = url.translate_connect_args(username='user')
         opts.update(url.query)
         keys = opts
-
         connect_args = {}
-        for param in ('ansi', 'unicode_results', 'autocommit'):
-            if param in keys:
-                connect_args[param] = util.asbool(keys.pop(param))
+        if 'autocommit' in keys:
+            connect_args['autocommit'] = util.asbool(keys.pop('autocommit'))
 
-        if 'odbc_connect' in keys:
-            connectors = [urllib.parse.unquote_plus(keys.pop('odbc_connect'))]
-        else:
-            dsn_connection = 'dsn' in keys or \
-                             ('host' in keys and 'database' not in keys)
-            if dsn_connection:
-                connectors = ['dsn=%s' % (keys.pop('host', '') or
-                                          keys.pop('dsn', ''))]
-            else:
-                connectors = [
-                    "DRIVER={%s}" %
-                    keys.pop('driver', self.pyodbc_driver_name),
-                    'System=%s' % keys.pop('host', ''),
-                    'DBQ=QGPL',
-                    "PKG=QGPL/DEFAULT(IBM),2,0,1,0,512"
-                ]
-                db_name = keys.pop('database', '')
-                if db_name:
-                    connectors.append("DATABASE=%s" % db_name)
-
-            user = keys.pop("user", None)
-            if user:
-                connectors.append("UID=%s" % user)
-                connectors.append("PWD=%s" % keys.pop('password', ''))
-            else:
-                connectors.append("trusted_connection=yes")
-
-            # if set to 'Yes', the ODBC layer will try to automagically convert
-            # textual data from your database encoding to your client encoding
-            # This should obviously be set to 'No' if you query a cp1253
-            # encoded database from a latin1 client...
-            if 'odbc_autotranslate' in keys:
-                connectors.append(
-                    "AutoTranslate=%s" %
-                    keys.pop("odbc_autotranslate"))
-
-            connectors.extend(['%s=%s' % (k, v) for k, v in keys.items()])
+        connectors = ['DSN=%s' % (keys.pop('host', '') or
+                                  keys.pop('dsn', ''))]
+
+        user = keys.pop("user", None)
+        if user:
+            connectors.append("UID=%s" % user)
+            connectors.append("PWD=%s" % keys.pop('password', ''))
+
+        connectors.extend(['%s=%s' % (k, v) for k, v in keys.items()])
         return [[";".join(connectors)], connect_args]
```

Comment:
The docs say this should return a connection string and keyword arguments to pass to the connect function of the dbapi, which in our case is PyODBC, so we need to return what https://github.com/mkleehammer/pyodbc/wiki/Module#connect wants. I think we could basically just pass 'Driver={IBM i Access ODBC driver}' as the connection string and pass all the connection options as a keyword dictionary. This would make the above code cleaner as we don't have to mess around with moving options from one dictionary (keys) to another (connect_args). Instead, we can use a dictionary comprehension as shown [here](https://stackoverflow.com/questions/3420122/filter-dict-to-contain-only-certain-keys) to only allow the user to set options which are supported.